src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -246,8 +193,18 @@ var TestRenderer = ReactFiberReconciler({
   useSyncScheduling: true,
 
   getPublicInstance(ref) {
-    const createNodeMock = ref.rootContainerInstance.createNodeMock;
-    return createNodeMock(ref);
+    if (typeof ref.text === 'string') {
+      return ref.text;
+    } else if (
+      typeof ref.rootContainerInstance === 'object' &&
+      typeof ref.rootContainerInstance.createNodeMock === 'function'
+    ) {
+      const createNodeMock = ref.rootContainerInstance.createNodeMock;
+      return createNodeMock(ref);
+    }
+
+    // this should not be possible
+    throw new Error('Attempted to getPublicInstance on an invalid ref.');
```

Comment:
I thought it was just `Instance` or `TextInstance`, though `Container` is also actually a possibility.

This was added for exhaustiveness because I couldn’t get flow to be okay without `else if` check just before this.